components/properties/src/script.rs

```diff
@@ -114,6 +134,17 @@ impl<'data> ScriptExtensions<'data> {
         }
     }
 
+    /// Return the `Script_Extensions` property value for this code point.
+    ///
+    /// If `code_point` has Script_Extensions, then return the Script codes in
+    /// the Script_Extensions. In this case, the Script property value
+    /// (normally Common or Inherited) is not included in the set.
+    ///
+    /// If c does not have Script_Extensions, then the one Script code is put
+    /// into the set and also returned.
+    ///
+    /// If c is not a valid code point, then the one UNKNOWN code is put into
+    /// the set and also returned.
     pub fn get_script_extensions_val(&self, code_point: u32) -> &ZeroSlice<Script> {
```

Comment:
I wonder if this should instead return a `ScriptExtensions` object that wraps the ZeroSlice, to hide implementation details and give us the opportunity to put helper methods on the return value.

Reply:
I guess the advantage to returning ZeroSlice is that we get all of ZeroSlice's impls for free, like iterators, getters, etc.  If we made a wrapper type, we'd need to add all those impls again.  However, I agree that a wrapper type would make it easier to change in the future.  @echeran can you open a follow-up issue so we can discuss this?